### ISSUE TITLE
Adding Essence Cost Reductions after Chargen Does Not Leave Essence Holes

### ISSUE THREAD

**Issue by DelnarErsike** [chummer5a/chummer5a#4536]
When a character receives an essence cost reduction after character creation (e.g. from Biocompatibility or houseruled Adapsin), there should be an Essence Hole left behind after the essence costs of appropriate 'ware have been reduced. Instead, the essence costs are just reduced, leaving no Essence Hole.

Side note: when attempting to fix this issue, see if you can do it without causing wonkiness when improvements are Re-applied. Otherwise, what could end up happening is that the Biocompatibility (or similar) impovement is removed, which increases Essence costs for 'ware, and when it is re-applied and Essence costs go back to normal, the character is left with an Essence hole equal to the total Essence cost reduction they had from Biocompatibility.

**Comment by chummer5a** [comment]
Is biocompatibility even supposed to be retroactive?

**Comment by ArchonMegalon** [comment]
No, it is not. At chargen arguably it makes sense, but after chargen it
should just discount new ware I think.

**Comment by DelnarErsike** [comment]
Chrome Flesh does not state that it is *not* retroactive whereas it does for Adapsin. It is also retroactive in HeroLab, and Missions FAQ states that for their house rules, it is retroactive but should leave an Essence hole.

**Comment by DelnarErsike** [comment]
Idea: have a stat that tracks the minimum Essence ever since leaving chargen and explicitly don't update it when reapplying improvements. If a retroactive Essence reduction is applied in Career mode, then an Essence Hole is generated (or increased) based on the difference between the minimum Essence ever and the current Essence after the modifier is applied.
